server/app/controllers/table_download_controller.rb

```diff
@@ -0,0 +1,43 @@
+class TableDownloadController < ApplicationController
+
+  @@table_resolvers = {
+    gene_interaction_results: [Resolvers::Genes, GeneInteractionsTsv],
+    drug_interaction_results: [Resolvers::Drugs, DrugInteractionsTsv],
+    gene_category_results: [Resolvers::Genes, GeneCategoriesTsv]
+  }
+
+  def download_table
+    table_name = params[:table_name].to_sym
+
+    if @@table_resolvers.has_key?(table_name)
+      res, mapper = @@table_resolvers[table_name]
+      stream_table(resolver: res, mapper: mapper)
+    else
+      head :bad_request
+    end
+  end
+
+  private
+  def stream_table(resolver: , mapper:)
+    variables = prepare_variables(params[:variables])
+    variables.delete('sortBy')
+    variables.transform_keys! { |k| GraphQL::Schema::Member::BuildType.underscore(k) }
+
+    headers.delete("Content-Length")
+    headers["Cache-Control"] = "no-cache"
+    headers["Content-Type"] = "text/csv"
```

Comment:
we might be able to make this `text/tab-separated-values` but idk if being more specific has any benefits